sklego/pandas_utils.py

```diff
@@ -11,45 +11,154 @@
 from scipy.ndimage.interpolation import shift
 
 
-def log_step(func=None, *, level=logging.INFO):
+def get_shape_delta(old_shape, new_shape):
+    diffs = [
+        ("+" if new > old else "") + str(new - old)
+        for new, old in zip(new_shape, old_shape)
+    ]
+
+    return f"delta=({', '.join(diffs)})"
+
+
+def log_step(
```

Comment:
Since you're addin the `log_*` updates. Might it also make sense to add the `print_*` variants? This might involve a small refactor but I'd love to also have something that is easy to hack around with in a jupyter notebook.

Reply:
It relates to https://github.com/koaning/scikit-lego/issues/371

Reply:
Yeah sure! Wouldn't adding a parameter to the call make more sense? That would make implementation easy and clean, and makes it less cumbersome (IMHO) to change if you move from notebooks to production.

Reply:
That's a fair consideraction that I hadn't thought of. One concern that I have is that the `log_step` function, as is, already has **a lot** of arguments and in my mind you'd need to change the decorator anyway for production because the logging configuration probably requires customisation.

Just in case I'll ping @MBrouns on this one too. I haven't done a lot of "pandas in prod" lately so I might be missing a vantage point.

Reply:
I've already added a logging parameter as well, which makes it easier to add your own handlers and formats (I'm using something similar at klm to have these logs in rich format)

Regarding the arguments: I agree. My current flow is to define a standard decorator per project (otherwise the logs will be very messy), and then it's probably less of a problem. I.e. you would define 
`project_log_step = log_step(...)` somewhere and import that log step from your own utils (or whatever) instead of the general one from sklego. But yeah let's see what Matthijs has to say :)

Reply:
That's quite neat!

For my use cases, it would need a small adaption still, because I generally only change the log_level within my code, i.e. I always use the same log_step and the only parameter that I change is the log_level. 

Maybe we could just allow log_level to be `print` or some negative value (since the loglevels are all int)?

Reply:
Now that I think of it though, it doesn't matter match whether it doesn't matter much whether I do `print_fn=logger.info` than `logger=logging.INFO`.

Reply:
I have a slight preference to assuming "print" as opposed to something-logging as a default. Mainly because my assumption is that most of the users of this tool will be folks working from a notebook. Notebooks appreciate prints more than loggers.

Reply:
Perfect, so let's go with matthijs' idea: `print_fn=print`. Will implement it somewhere this weekend

Reply:
Done!